app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/card/CardViewHolder.kt

```diff
@@ -1,67 +1,77 @@
 package de.tum.`in`.tumcampusapp.component.ui.overview.card
 
 import android.app.Activity
+import android.content.Context
 import android.content.Intent
 import android.support.v7.widget.PopupMenu
 import android.support.v7.widget.RecyclerView
 import android.view.Gravity
 import android.view.MenuItem
 import android.view.View
+import android.widget.ImageView
 import de.tum.`in`.tumcampusapp.R
 import de.tum.`in`.tumcampusapp.component.other.navigation.NavigationManager
 import de.tum.`in`.tumcampusapp.component.other.settings.UserPreferencesActivity
+import de.tum.`in`.tumcampusapp.component.ui.overview.CardInteractionListener
 import de.tum.`in`.tumcampusapp.utils.Const
 
-open class CardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView),
-        View.OnClickListener, View.OnLongClickListener, PopupMenu.OnMenuItemClickListener {
+open class CardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
     var currentCard: Card? = null
-    private val activity: Activity
+
+    private val context: Context by lazy { itemView.context }
+    private val activity: Activity by lazy { context as Activity }
+    private val listener: CardInteractionListener? by lazy { context as? CardInteractionListener }
 
     init {
-        itemView.setOnClickListener(this)
-        itemView.setOnLongClickListener(this)
-        activity = itemView.context as Activity
+        itemView.setOnClickListener {
+            val destination = currentCard?.getNavigationDestination()
+            destination?.let { dest ->
+                NavigationManager.open(activity, dest)
+            }
+        }
+
+        val moreIcon = itemView.findViewById<ImageView>(R.id.cardMoreIcon)
+        moreIcon?.setOnClickListener { openOptionsPopup(it) }
     }
 
-    override fun onClick(v: View) {
-        val destination = currentCard?.getNavigationDestination()
-        destination?.let {
-            NavigationManager.open(activity, it)
+    private fun openOptionsPopup(anchorView: View) {
+        val card = currentCard ?: return
+        val menuId = if (card.hasSettings) {
```

Comment:
I think a boolean property isn't the right abstraction for this...
If we want to go down this route, we probably should let the cards itself specify it's options popup layout. A boolean property just obscures what options the card actually needs to support.

Reply:
You’re right – I can change it so that every card provides its own menu: 
`menuInflater.inflate(currentCard.optionsMenu, menu)`

Is that what you had in mind, @pfent?

Reply:
Something like that, yes

Reply:
We could then even make those menus a bit more useful eventually - good thinking :+1: